### PR TITLE
Enc common555

### PR DESCRIPTION
**Pull request by andrey-norkin**
This PR continues separation of the encoder library into encoder and common. In particular, the separation has been done for inter prediction and a number of intrinsics.

### CODE REVIEW

File: Source/Lib/Common/Codec/common_dsp_rtcd.c [hunk 1]
```diff
@@ -368,4 +377,97 @@ void setup_common_rtcd_internal(CPU_FLAGS flags) {
...
+    SET_AVX2_AVX512(eb_av1_wiener_convolve_add_src,
+                    eb_av1_wiener_convolve_add_src_c,
+                    eb_av1_wiener_convolve_add_src_avx2,
+                    eb_av1_wiener_convolve_add_src_avx512);
+
+    eb_av1_convolve_2d_copy_sr = eb_av1_convolve_2d_copy_sr_c;
+    if (flags & HAS_AVX2) eb_av1_convolve_2d_copy_sr = eb_av1_convolve_2d_copy_sr_avx2;
+
+    eb_av1_convolve_2d_scale = eb_av1_convolve_2d_scale_c;
+    //if (flags & HAS_SSE4_1) eb_av1_convolve_2d_scale = eb_av1_convolve_2d_scale_sse4_1;
```
Comment (hassount) [inline Source/Lib/Common/Codec/common_dsp_rtcd.c]: are these commented out by design?

File: Source/Lib/Common/Codec/common_dsp_rtcd.c [hunk 2]
```diff
@@ -368,4 +377,97 @@ void setup_common_rtcd_internal(CPU_FLAGS flags) {
...
+    eb_av1_highbd_jnt_convolve_2d_copy = eb_av1_highbd_jnt_convolve_2d_copy_c;
+    if (flags & HAS_AVX2)
+        eb_av1_highbd_jnt_convolve_2d_copy = eb_av1_highbd_jnt_convolve_2d_copy_avx2;
+    eb_av1_highbd_convolve_y_sr = eb_av1_highbd_convolve_y_sr_c;
+    if (flags & HAS_AVX2) eb_av1_highbd_convolve_y_sr = eb_av1_highbd_convolve_y_sr_avx2;
+    eb_av1_highbd_convolve_2d_sr = eb_av1_highbd_convolve_2d_sr_c;
+    if (flags & HAS_AVX2) eb_av1_highbd_convolve_2d_sr = eb_av1_highbd_convolve_2d_sr_avx2;
+
+    eb_av1_highbd_convolve_2d_scale = eb_av1_highbd_convolve_2d_scale_c;
+    //if (flags & HAS_SSE4_1) eb_av1_highbd_convolve_2d_scale = eb_av1_highbd_convolve_2d_scale_sse4_1
```
Comment (hassount) [inline Source/Lib/Common/Codec/common_dsp_rtcd.c]: are these commented out by design?

File: Source/Lib/Common/Codec/common_dsp_rtcd.c [hunk 3]
```diff
@@ -368,4 +377,97 @@ void setup_common_rtcd_internal(CPU_FLAGS flags) {
...
+                    eb_av1_jnt_convolve_2d_copy_avx2,
+                    eb_av1_jnt_convolve_2d_copy_avx512);
+    SET_AVX2_AVX512(eb_av1_jnt_convolve_x,
+                    eb_av1_jnt_convolve_x_c,
+                    eb_av1_jnt_convolve_x_avx2,
+                    eb_av1_jnt_convolve_x_avx512);
+    SET_AVX2_AVX512(eb_av1_jnt_convolve_y,
+                    eb_av1_jnt_convolve_y_c,
+                    eb_av1_jnt_convolve_y_avx2,
+                    eb_av1_jnt_convolve_y_avx512);
```
Comment (hassount) [inline Source/Lib/Common/Codec/common_dsp_rtcd.c]: @ttrigui 
Do we need to protect these under the AVX512 flag?

File: Source/Lib/Encoder/ASM_SSE4_1/EbBlend_sse4.h [hunk 4]
```diff
@@ -1,390 +0,0 @@
-/*
```
Comment (hassount) [inline Source/Lib/Encoder/ASM_SSE4_1/EbBlend_sse4.h]: can we delete this file if it's empty now

File: Source/Lib/Common/Codec/common_dsp_rtcd.c (hunk 3 above)
Comment (ttrigui) [reply Source/Lib/Common/Codec/common_dsp_rtcd.c]: no we do not need to, SET_FUNCTIONS is already protected by the AVX512 flag. It sets AVX512 to AVX2 if AVX512 is disabled.

**Comment by hassount** [comment]
Confirmed bit stream exactness for -enc-mode 0,4,8 with the master. Can we add a good description to this PR address the comments and squash the commits so that we can merge this?

File: Source/Lib/Common/Codec/common_dsp_rtcd.c (hunk 3 above)
Comment (hassount) [reply Source/Lib/Common/Codec/common_dsp_rtcd.c]: Thank you @ttrigui for the confirmation

File: Source/Lib/Encoder/ASM_SSE4_1/EbBlend_sse4.h (hunk 4 above)
Comment (andrey-norkin) [reply Source/Lib/Encoder/ASM_SSE4_1/EbBlend_sse4.h]: The file has been moved to common and does not exist in the encoder lib any more. I think it is an artifact of how github is showing it.

File: Source/Lib/Common/Codec/common_dsp_rtcd.c (hunk 2 above)
Comment (andrey-norkin) [reply Source/Lib/Common/Codec/common_dsp_rtcd.c]: This is how it was in the aom_dsp_rtcd.c.  Can be removed though.

File: Source/Lib/Common/Codec/common_dsp_rtcd.c (hunk 1 above)
Comment (andrey-norkin) [reply Source/Lib/Common/Codec/common_dsp_rtcd.c]: This is how it was in the aom_dsp_rtcd.c.  Can be removed though.

**Pull request merged**